model_compression_toolkit/core/common/quantization/set_node_quantization_config.py

```diff
@@ -329,36 +337,137 @@ def filter_qc_options_with_manual_bit_width(
+
+    return base_config, node_qc_options_list
+
+
+def filter_weights_qc_options_with_manual_bit_width(
+        node: BaseNode,
+        node_qc_options_list: List[OpQuantizationConfig],
+        base_config: OpQuantizationConfig,
+        weights_manual_bit_width_override: Optional[list],
+        mixed_precision_enable: bool) -> Tuple[OpQuantizationConfig, List[OpQuantizationConfig]]:
+    """
+    Update the weights quantization configurations for a node, allowing manual bit-width overrides if specified.
+
+    Args:
+        node (BaseNode): A node to set quantization configuration candidates to.
+        node_qc_options_list (List[OpQuantizationConfig]): List of quantization configs for the node.
+        base_config (OpQuantizationConfig): Base quantization config for the node.
+        weights_manual_bit_width_override (Optional[Dict]): Specifies a custom bit-width to override the node's weights bit-width.
+        mixed_precision_enable (bool): Whether mixed precision is enabled.
+
+    Returns:
+        Tuple[OpQuantizationConfig, List[OpQuantizationConfig]]: The updated base configuration and the filtered list of quantization configs.
+    """
+    if not weights_manual_bit_width_override:
+        return base_config, node_qc_options_list
+
+    # Filter node_qc_options_list to retain only the options with weights bits equal to weights_manual_bit_width_override.
+    node_qc_options_weights_list = []
+    override_attr, override_bitwidth = [], []
+
+    for weights_manual_bit_width in weights_manual_bit_width_override:
+        override_attr.append(weights_manual_bit_width[1])
+        override_bitwidth.append(weights_manual_bit_width[0])
+
+    node_qc_options_weights_list = copy.deepcopy(node_qc_options_list)
+    for attr, bitwidth in zip(override_attr, override_bitwidth):
+        for op_cfg in node_qc_options_list:
+            if op_cfg in node_qc_options_weights_list:
+                weights_attrs = op_cfg.attr_weights_configs_mapping.keys()
+                if attr in weights_attrs:
+                    for weights_attr in weights_attrs:
+                        if attr == weights_attr and op_cfg.attr_weights_configs_mapping.get(attr).weights_n_bits != bitwidth:
+                            node_qc_options_weights_list.remove(op_cfg)
+                else:
+                    node_qc_options_weights_list.remove(op_cfg)
+
+    if len(node_qc_options_weights_list) == 0:
+        Logger.critical(f"Manually selected weights bit-width {weights_manual_bit_width_override} is invalid for node {node}.")
+    else:
+        # Update the base_config to one of the values from the filtered node_qc_options_list.
+        # First, check if a configuration similar to the original base_config but with weights bits equal to weights_manual_bit_width_override exists.
+        # If it does, use it as the base_config. If not, choose a different configuration from node_qc_options_list.
+        updated_base_config = copy.deepcopy(base_config)
```

Comment:
use `base_config.clone_and_edit` (see how it is used in the activation manual selection method)